`InstallGitLFS.install()` should support alternate URLs for `git-lfs` GitHub release asset URL

roynos opened the pull request:
#### 3c70a091d1fb016fce5a5c2774cc67fc4282b4fd
<pre>
`InstallGitLFS.install()` should support alternate URLs for `git-lfs` GitHub release asset URL
<a href="https://bugs.webkit.org/show_bug.cgi?id=246306">https://bugs.webkit.org/show_bug.cgi?id=246306</a>
rdar://problem/101002204

Reviewed by Jonathan Bedard.

Users can define a `mirror_resolver_func` callback that takes a URL. The callback function is responsible for providing the alternate URL.

* Tools/Scripts/libraries/webkitscmpy/setup.py: version bump.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: version bump
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_git_lfs.py:
(InstallGitLFS.url): handle `mirror_resolver_func`
(InstallGitLFS.install): handle `mirror_resolver_func`
(InstallGitLFS.parser): handle `mirror_resolver_func`
(InstallGitLFS.main): handle `mirror_resolver_func`

Canonical link: <a href="https://commits.webkit.org/255413@main">https://commits.webkit.org/255413@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b1150aa54ea445afeefa562cfb7700154a01fa17

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92468 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1691 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23053 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/102224 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/162676 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96469 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1687 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30064 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/84886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/98375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98131 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/1126 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78971 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/84886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/96044 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/83047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/82732 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/84886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/36475 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/16671 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/91514 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34251 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/17844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3747 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38119 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/40463 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40028 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/36994 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_git_lfs.py:
```diff
@@ -121,6 +124,13 @@ def parser(cls, parser, loggers=None):
             dest='configure',
             default=None,
         )
+        parser.add_argument(
```
I'm a bit unclear how a parser helps us here. `mirror_resolver_func` needs to be callable, but you can only define a string via the command line, right? So how does one use the `--mirror-resolver-func` option, then?

roynos replied:
done.

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_git_lfs.py:
```diff
@@ -121,6 +124,10 @@ def parser(cls, parser, loggers=None):
             default=None,
         )
+        parser.set_defaults(
+            mirror_resolver_func=None,
```
I am fine with this as-is, although usually I've done this via `kwargs` on `main` (which is exactly the reason it's `kwargs` in the first place, the parent `main` passes some arguments that not all child-mains use.

roynos committed:
`InstallGitLFS.install()` should support alternate URLs for `git-lfs` GitHub release asset URL
https://bugs.webkit.org/show_bug.cgi?id=246306
rdar://problem/101002204

Reviewed by Jonathan Bedard.

Users can define a `mirror_resolver_func` callback that takes a URL. The callback function is responsible for providing the alternate URL.

* Tools/Scripts/libraries/webkitscmpy/setup.py: version bump.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: version bump
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_git_lfs.py:
(InstallGitLFS.url): handle `mirror_resolver_func`
(InstallGitLFS.install): handle `mirror_resolver_func`
(InstallGitLFS.parser): handle `mirror_resolver_func`
(InstallGitLFS.main): handle `mirror_resolver_func`

Canonical link: https://commits.webkit.org/255413@main
